### PR TITLE
Check renameat2() is available on Linux

### PR DESCRIPTION
**Pull request by spevans**
Add a check that the system call is defined and the kernel version is high enough.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@swift-ci test linux

File: CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h [hunk 1]
```diff
@@ -594,10 +595,21 @@ _stat_with_btime(const char *filename, struct stat *buffer, struct timespec *bti
 #endif // __NR_statx
 
 static unsigned int const _CF_renameat2_RENAME_EXCHANGE = 1 << 1;
+#ifdef SYS_renameat2
+static _Bool const has_sys_renameat2 = 1;
```
Comment (millenomi) [inline CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h]: Please prefix this? Eg _CFHasRenameat2.

Comment (millenomi) [reply CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h]: (And below, obviously.)

Comment (spevans) [reply CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h]: fixed

**Comment by millenomi** [comment]
Approved.

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Pull request merged**